sequential/seq2pat.py

```diff
@@ -283,12 +294,36 @@ class Seq2Pat:
         sequences but no constraints are used to run the mining efficiently and practically.
         Power users can choose to drop this constraint by setting it to be None or increase the maximum span
         as the system has resources to support.
+    batch_size: Optional[int]
+        When batch_size is set, program runs Seq2Pat on batches of sequences instead of on the entire set of
+        sequences for improving scalability. Each batch contains `batch_size` sequences as a random sample of
+        entire set. A mining task will run on each batch with a reduced minimum row count (min_frequency) threshold.
+        Please refer to description of discount_factor parameter for how min_frequency is reduced.
+        Resulted patterns will be aggregated from the mining results of each batch by calculating the sum of
+        the occurrences. Finally the original minimum row count threshold is applied to the patterns after aggregation.
+    n_jobs: int
+        n_jobs defines the number of processes (n_jobs=2 by default) that are used when mining tasks are applied
+        on batches in parallel. If -1 all CPUs are used. If -2, all CPUs but one are used.
+    seed: int
+        Random seed to make sequences uniformly distributed among batches.
+    discount_factor: float
+        A discount factor is used to reduce the minimum row count (min_frequency) threshold when Seq2Pat is applied
+        on a batch. When min_frequency is a integer, mining task can only be run on the entire set.
+        When min_frequency is a float, mining can be run on batches, with new threshold being defined by
+        max(min_frequency * discount_factor, 1.0/num_rows). num_rows is the number of sequences of one batch.
+        Final results will be based on the aggregation of patterns from each batch by calculating the sum of
+        the occurrences. Theoretically there is a chance that the batching results will be different from non-batching
+        results. But a small discount_factor parameter will make the chance to be minimal and thus we have the same
+        results as running on entire set in practices. A small value of discount_factor is thus recommended.
+        discount_factor=0.2 by default.
     """
 
-    def __init__(self, sequences: List[list], max_span: Optional[int] = 10):
+    def __init__(self, sequences: List[list], max_span: Optional[int] = 10, batch_size=None,
```

Comment:
shall we consider re-ordering parameters so that batch stuff is together 

batch_size, discoun_factor, n_jobs, seed

Reply:
So the Default behavior is None? Can you say more, why? I thought this was to make "naive" users not to run into memory issues, similar to fixing max_span=10 so that it runs fast. Shall we consider enforcing batch by default, or is it too dramatic? 

Btw, what happens if Batch_size is larger than num_rows in the original data? I assume we handle that inside (and should have a unit test for that edge case, if not already)

Reply:
Done with the reordering of parameters. It makes sense to put batch parameters together.

Regarding the default behavior for batch, I assume the dataset might not be in a size of a few millions in many cases, so running on entire set won't be an issue then and it can be faster. For example, in one analysis, the data is 10000, if I set batch_size=10000 as default, it is slower than not using batch due to the introduced operations, e.g. shuffle, discount factor... The runtime benefit will be obvious only when we deal with large dataset. So I think we can make default behavior to be single thread for now.

And yes, if batch_size is larger than num_rows, we can still get right results. I have a test to cover this.